internal/infrastructure/manager.go

```diff
@@ -34,6 +34,8 @@ func NewManager(cfg *config.Server) (*Manager, error) {
 		ns, found := os.LookupEnv("ENVOY_GATEWAY_NAMESPACE")
 		if found {
 			mgr.Infra.Namespace = ns
```

Comment:
nit: would have preferred if this logic was living in `internal/infrastructure/kubernetes`